perf-test/clj/reitit/opensensors_routing_test.clj

```diff
@@ -0,0 +1,563 @@
+                       (->route-name)))]
+    (-> (slurp url)
+        (json/parse-string true)
+        (->> :paths
+             (mapv (fn [[p v]] [(-> p name (str/replace #"\{(.*?)\}" ":$1") (->> (str "/"))) (->endpoint v)]))))))
+
+(defn valid-urls [router]
+  (->>
+    (for [name (reitit/route-names router)
+          :let [match (reitit/match-by-name router name)
+                params (if (reitit/partial-match? match)
+                         (-> match :required (zipmap (range))))]]
+      (:path (reitit/match-by-name router name params)))
+    (into [])))
+
+(comment
+  (swagger->routes "https://api.opensensors.io/doc" false))
+
+(defn bench-routes [routes f]
+  (let [router (reitit/router routes)
+        urls (valid-urls router)
+        random-url #(rand-nth urls)
+        log-time #(let [now (System/nanoTime)] (%) (- (System/nanoTime) now))
+        total 10000
+        dropped (int (* total 0.45))]
+    (mapv
+      #(let [times (->> (range total)
+                        (mapv
+                          (fn [_]
+                            (let [now (System/nanoTime)
+                                  result (f %)
+                                  total (- (System/nanoTime) now)]
+                              (assert result)
+                              total)))
+                        (sort)
+                        (drop dropped)
+                        (drop-last dropped))
+             avg (int (/ (reduce + times) (count times)))]
+         [% avg]) urls)))
+
+(defn bench [routes no-paths?]
+  (let [routes (mapv (fn [[path name]]
+                       (if no-paths?
+                         [(str/replace path #"\:" "") name]
+                         [path name])) routes)
+        router (reitit/router routes)]
+    (doseq [[path time] (bench-routes routes #(reitit/match-by-path router %))]
+      (println path "\t" time))))
+
+(defn bench [routes no-paths?]
+  (let [routes (mapv (fn [[path name]]
+                       (if no-paths?
+                         [(str/replace path #"\:" "") name]
+                         [path name])) routes)
+        router (reitit/router routes)]
+    (doseq [[path time] (bench-routes routes #(reitit/match-by-path router %))]
+      (println path "\t" time))))
+;;
+;; Perf tests
+;;
+
+(def handler (constantly {:status 200, :body "ok"}))
+
+(def opensensors-routes
+  [["/v2/whoami" {:handler handler, :name :test/route1}]
+   ["/v2/users/:user-id/datasets" {:handler handler, :name :test/route2}]
+   ["/v2/public/projects/:project-id/datasets" {:handler handler, :name :test/route3}]
+   ["/v1/public/topics/:topic" {:handler handler, :name :test/route4}]
+   ["/v1/users/:user-id/orgs/:org-id" {:handler handler, :name :test/route5}]
+   ["/v1/search/topics/:term" {:handler handler, :name :test/route6}]
+   ["/v1/users/:user-id/invitations" {:handler handler, :name :test/route7}]
+   #_["/v1/orgs/:org-id/devices/:batch/:type" {:handler handler, :name :test/route8}]
```

Comment:
There are conflicting routes and pedestals breaks with those. Also, pedestal fails on non-conflicting routes. Will write issue there.